### PR TITLE
chore(Promotions): Migrate #85 - Add styling for promotion card

### PR DESCRIPTION
**Pull request by andrewzulaybar**
## Description

This change just adds styling to the promotion cards to implement the hi-fi designs. Currently, I just hardcoded the image URL and schedule for each promotion so they're all showing up as the same but this should be updated to use the data from the backend.

## Photos
Screen Shot 2021-01-30 at 11 21 27 PM

### CODE REVIEW

File: src/components/promotion/PromotionDetails.tsx [hunk 1]
```diff
@@ -1,42 +1,181 @@
...
+  /**
+   * Returns display text for age of promotion.
+   *
+   * - If dateAdded < 1 day, displays "X hours ago"
+   * - If 1 day <= dateAdded < 1 week, displays "X days ago"
+   * - If 1 week <= dateAdded < 1 month, displays "X weeks ago"
+   * - If 1 month <= dateAdded < 1 year, displays "X months ago"
+   * - If 1 year <= dateAdded, displays "X years ago"
+   */
+  const promotionAge = (dateAdded: string) => {
```
Comment (ericrkuo) [inline src/components/promotion/PromotionDetails.tsx]: I wonder if we can leverage an NPM package for this, maybe something like https://www.npmjs.com/package/time-ago? 

Regardless, this is really cool

File: src/components/promotion/PromotionImage.tsx [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+  image: {
+    borderRadius: 15,
+    minHeight: '100%',
+    objectFit: 'cover',
+    width: 100,
+  },
+};
+
+export default function PromotionImage({
+  // TODO: remove default image
```
Comment (ericrkuo) [inline src/components/promotion/PromotionImage.tsx]: Maybe we can link the issue to this comment as well? https://github.com/ubclaunchpad/PromoPal-backend/issues/101?

**Review by ericrkuo — APPROVED** [review]

File: src/components/promotion/PromotionDetails.tsx (hunk 1 above)
Comment (andrewzulaybar) [reply src/components/promotion/PromotionDetails.tsx]: Oooh good idea, what do you think of https://date-fns.org/? It seems like it has a lot of support and we could potentially use their other functions if we have needs for it in the future

Usage ex:
```
formatDistance(subDays(new Date(), 3), new Date())
```

File: src/components/promotion/PromotionImage.tsx (hunk 2 above)
Comment (andrewzulaybar) [reply src/components/promotion/PromotionImage.tsx]: Ahhh yes I'll add this thanks!

File: src/components/promotion/PromotionDetails.tsx (hunk 1 above)
Comment (ericrkuo) [reply src/components/promotion/PromotionDetails.tsx]: Oooh that looks even better, let's go w that then!

File: src/components/promotion/PromotionDetails.tsx [hunk 3]
```diff
@@ -1,42 +1,139 @@
...
+
+      const hourNum = parseInt(hour);
+      const minuteNum = parseInt(minute);
+
+      let formattedTime = `${hourNum === 12 ? hourNum : hourNum % 12}`;
+      if (minuteNum !== 0) {
+        formattedTime += `:${minuteNum}`;
+      }
+      formattedTime += hourNum < 12 ? ' AM' : ' PM';
+      return formattedTime;
```
Comment (ericrkuo) [inline src/components/promotion/PromotionDetails.tsx:78]: Does `date-fns` have something similar to this operation?

**Pull request merged**